airflow/utils/orm_event_handlers.py

```diff
@@ -46,6 +46,7 @@ def connect(dbapi_connection, connection_record):
         def set_sqlite_pragma(dbapi_connection, connection_record):
             cursor = dbapi_connection.cursor()
             cursor.execute("PRAGMA foreign_keys=ON")
+            cursor.execute("PRAGMA journal_mode=WAL")
```

Comment:
This is essentially the change, everything else is removing code that isn't needed anymore!